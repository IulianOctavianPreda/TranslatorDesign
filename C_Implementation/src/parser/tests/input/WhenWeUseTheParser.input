int main() {
    int i; 
    int _n; 
    int t1;
    int t2;
    int nextTerm;

    i = 1;
    _n = 100; 
    t1 = 0; 
    t2 = 1;
    while (i <= _n) {
        nextTerm = t1 + t2;
        t1 = t2;
        t2 = nextTerm;
        i = i + 1;
    }
    return 0;
}
