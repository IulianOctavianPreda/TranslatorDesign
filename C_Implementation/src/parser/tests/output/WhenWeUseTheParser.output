Type (3)
  TYPE: int
