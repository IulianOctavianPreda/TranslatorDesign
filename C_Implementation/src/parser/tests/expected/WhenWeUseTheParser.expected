Program (1)
  Program (1)
  FnDecl (1)
    Type (1)
      TYPE: int
    Id (1)
      ID: main
    Parameters (1)
      LRB
      RRB
    Block (1)
      LB
      DeclList (1)
        DeclList (1)
          DeclList (1)
            DeclList (1)
              DeclList (1)
                DeclList (1)
                VarDecl (2)
                  Type (2)
                    TYPE: int
                  Id (2)
                    ID: i
                  SEMICOL
              VarDecl (3)
                Type (3)
                  TYPE: int
                Id (3)
                  ID: _n
                SEMICOL
            VarDecl (4)
              Type (4)
                TYPE: int
              Id (4)
                ID: t1
              SEMICOL
          VarDecl (5)
            Type (5)
              TYPE: int
            Id (5)
              ID: t2
            SEMICOL
        VarDecl (6)
          Type (6)
            TYPE: int
          Id (6)
            ID: nextTerm
          SEMICOL
      DeclList (8)
        DeclList (8)
          DeclList (8)
            DeclList (8)
              DeclList (8)
                DeclList (8)
                  StmtList (8)
                  Stmt (8)
                    Id (8)
                      ID: i
                    EQ
                    Exp (8)
                      Atom (8)
                        INT: 1
                    SEMICOL
                Stmt (9)
                  Id (9)
                    ID: _n
                  EQ
                  Exp (9)
                    Atom (9)
                      INT: 100
                  SEMICOL
              Stmt (10)
                Id (10)
                  ID: t1
                EQ
                Exp (10)
                  Atom (10)
                    INT: 0
                SEMICOL
            Stmt (11)
              Id (11)
                ID: t2
              EQ
              Exp (11)
                Atom (11)
                  INT: 1
              SEMICOL
          Stmt (12)
            WHILE
            LRB
            Exp (12)
              Exp (12)
                Atom (12)
                  Id (12)
                    ID: i
              LE_OP
              Exp (12)
                Atom (12)
                  Id (12)
                    ID: _n
            RRB
        Stmt (18)
          RETURN
          Exp (18)
            Atom (18)
              INT: 0
          SEMICOL
      RB
